modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/FunctionObjectValidationRuleEngineImpl.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.object.internal.validation.rule;
+
+import com.liferay.object.constants.ObjectValidationRuleConstants;
+import com.liferay.object.internal.configuration.FunctionObjectActionExecutorImplConfiguration;
+import com.liferay.object.validation.rule.ObjectValidationRuleEngine;
+import com.liferay.osgi.util.configuration.ConfigurationFactoryUtil;
+import com.liferay.portal.catapult.PortalCatapult;
+import com.liferay.portal.configuration.metatype.bnd.util.ConfigurableUtil;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.util.Http;
+
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Mateus Santana
+ */
+@Component(
+	configurationPid = "com.liferay.object.internal.configuration.FunctionObjectValidationRuleEngineImplConfiguration",
+	factory = "com.liferay.object.internal.configuration.FunctionObjectValidationRuleEngineImplConfiguration",
+	service = ObjectValidationRuleEngine.class
+)
+public class FunctionObjectValidationRuleEngineImpl
+	implements ObjectValidationRuleEngine {
+
+	@Override
+	public Map<String, Object> execute(
+		Map<String, Object> inputObjects, String script) {
+
+		_portalCatapult.launch(
+			_companyId, Http.Method.POST,
+			_functionObjectActionExecutorImplConfiguration.
+				oAuth2ApplicationExternalReferenceCode(),
+			_jsonFactory.createJSONObject(
+				_jsonFactory.looseSerialize(
+					notificationContext, "notificationRecipientSettings",
```

Comment:
This `notificationContext` is the JSON Payload, right?
How do you discover the right way to mount this payload?

Reply:
> This notificationContext is the JSON Payload, right?

right!

> How do you discover the right way to mount this payload?

the payload can vary for each client extension, in this case a choose this object to be the payload,

but I made a mistake, I should pass a DTO or an object that contains more than one DTO,

so for your client extension, you must pass the objectEntry DTO, if you need to pass other data make sure that is a DTO defined in the headless layer.

Reply:
Alright!
In this case, I am allowing the client to validate this entry by sending the Object Entry. Am I wrong?
If so, the validation should be made as the client needs.

Reply:
you're right, you just need to pass the object entry dto and the client will validate in their microsservice.

Reply:
I found [this example](https://github.com/liferay/liferay-portal/blob/a73ad2bf2a0c3e5348835f0995a5266462c87982/modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java#L1028) for mount the payload.